app/main.py

```diff
@@ -38,6 +39,35 @@
 import setup_pf
 
 
+class AdminEnv(object):
+    """Template variables for admin pages."""
+
+    def __init__(self, request):
+        self.request = request
+        self.user = users.get_current_user()
+        self.logout_url = users.create_logout_url(self.request.url)
+
+    @property
+    def repo_options(self):
+        """This is different from env.repo_options because this contains all
+        repositories including deactivated ones.
+
+        This is defined as a property so that it is evaluated lazily only
+        when necessary.
+        """
+        try:
+            return [
+                utils.Struct(
+                    repo=repo,
+                    url=utils.get_repo_url(self.request, repo) + '/admin')
+                for repo in sorted(model.Repo.list())]
+        except:
+            # Logs the exception here because exceptions thrown during template
+            # variable evaluation is silently ignored.
+            logging.exception('exception')
```

Comment:
Are you sure you want to log the string 'exception'? and not the actual exception?